controllers/volumereplication_controller.go

```diff
@@ -317,6 +325,70 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForVolumeReplicationClass(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplication(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationClass(logger logr.Logger) error {
+
+	isVolumeReplicationClassCrdExists := false
+	for ! isVolumeReplicationClassCrdExists {
+		vrcObj := &replicationv1alpha1.VolumeReplicationClassList{}
+		err := r.Client.List(context.TODO(), vrcObj)
+		if err != nil {
+			if meta.IsNoMatchError(err){
+				logger.Info("Resource", "VolumeReplicationClass", "does not exist")
+				time.Sleep(5 * time.Second)
+			} else if ! strings.Contains(err.Error(), "cache is not started"){
+				logger.Error(err, "Got an unexpected error while waiting for", "Resource", "VolumeReplicationClass")
+				return err
+			} else {
+				isVolumeReplicationClassCrdExists = true
+			}
+		} else {
+			isVolumeReplicationClassCrdExists = true
+		}
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplication(logger logr.Logger) error {
+
+	isVolumeReplicationCrdExists := false
+	for ! isVolumeReplicationCrdExists {
+		vrOBJ := &replicationv1alpha1.VolumeReplicationList{}
+		err := r.Client.List(context.TODO(), vrOBJ)
+		if err != nil {
+			if meta.IsNoMatchError(err){
+				logger.Info("Resource", "VolumeReplication", "does not exist")
+				time.Sleep(5 * time.Second)
+			} else if ! strings.Contains(err.Error(), "cache is not started"){
```

Comment:
can you please add a comment here?
will this happen on all SDK versions or specific SDK versions?
are we considering `cache is not started` as the success message here and why?

Reply:
> are we considering `cache is not started` as the success message here and why?
>
if we get `cache is not started` error, that means we have the wanted crd in the cluster

Reply:
> can you please add a comment here?
>
what will be the comment on? why we put the `cache is not started` there?

Reply:
if we get `cache is not started` it doesn't mean the CRD is present rather it's a different error? can we just consider this as a normal error message to let the operator restart? or use a different client as mentioned at https://github.com/openshift-knative/serverless-operator/pull/658 https://github.com/jaegertracing/jaeger-operator/issues/809#issue-532751170

Reply:
I tried what you suggested but it didn't work well.
I used [unstructured](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured) module and it solves the `cache is not started` problem